agent/src/main/scala/za/co/absa/atum/agent/core/MeasurementProcessor.scala

```diff
@@ -0,0 +1,105 @@
+package za.co.absa.atum.agent.core
+
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.{DecimalType, LongType, StringType}
+import org.apache.spark.sql.{Column, DataFrame, Dataset, Row}
+import za.co.absa.atum.agent.model._
+import za.co.absa.atum.agent.utils.controlmeasure.ControlMeasureUtils
+
+trait MeasurementProcessor {
+
+  type MeasurementFunction = DataFrame => String
+
+  def getFunction(
+    dataset: Dataset[Row],
+    measurement: Measurement
+  ): MeasurementFunction
+
+}
+
+class MeasurementProcessorImplementation extends MeasurementProcessor {
```

Comment:
Instead of this implementation I would - with the companion object - define classes inheriting from `MeasurementProcessor`, one for  each measurement class.
If there would be an enumeration for measurement types, this would ensure at compile time that for each `Measurement` exists a `MeasurementProcessor`.